controllers/config/defaults.go

```diff
@@ -59,16 +62,16 @@ const (
 // Default ResourceRequirements
 
 var (
-	APIServerResourceRequirements         = createResourceRequirement("250m", "500Mi", "500m", "1Gi")
-	PersistenceAgentResourceRequirements  = createResourceRequirement("120m", "500Mi", "250m", "1Gi")
-	ScheduledWorkflowResourceRequirements = createResourceRequirement("120m", "100Mi", "250m", "250Mi")
-	ViewerCRDResourceRequirements         = createResourceRequirement("120m", "100Mi", "250m", "500Mi")
-	MariaDBResourceRequirements           = createResourceRequirement("300m", "800Mi", "1", "1Gi")
-	MinioResourceRequirements             = createResourceRequirement("200m", "100Mi", "250m", "1Gi")
-	MlPipelineUIResourceRequirements      = createResourceRequirement("100m", "256Mi", "100m", "256Mi")
+	APIServerResourceRequirements         = createResourceRequirement(resource.MustParse("250m"), resource.MustParse("500Mi"), resource.MustParse("500m"), resource.MustParse("1Gi"))
```

Comment:
I'm not sure if this is idiomatic go or not. Feel free to suggest a better way.

Reply:
change set can be shorter by just changing the struct builder: 

```golang
func createResourceRequirement(RequestsCPU string, RequestsMemory string, LimitsCPU string, LimitsMemory string) dspav1alpha1.ResourceRequirements {
	return dspav1alpha1.ResourceRequirements{
		Requests: &dspav1alpha1.Resources{
			CPU:    resource.MustParse(RequestsCPU),
			Memory: resource.MustParse(RequestsMemory),
		},
		Limits: &dspav1alpha1.Resources{
			CPU:    resource.MustParse(LimitsCPU),
			Memory: resource.MustParse(LimitsMemory),
		},
	}
}
```

Reply:
From my reading of the [doc](https://pkg.go.dev/k8s.io/apimachinery/pkg/api/resource) and the examples provided, it seems unsafe to put the `MustParse` call into the function. Doc says:
"MustParse turns the given string into a quantity or panics; for tests or other cases where you know the string is valid." Note that it doesn't return an err -- it just panics.

I admit that it's unlikely that we'll ever call the function with anything other than a valid Quantity string, but forcing callers to call MustParse themselves felt a little safer to me.

If you still think it's better to change it, I'll change it.